ui/analyse/src/serverSideUnderboard.ts

```diff
@@ -30,7 +30,23 @@ export default function (element: HTMLElement, ctrl: AnalyseCtrl) {
     unselect = (chart: Highcharts.ChartObject) => chart.getSelectedPoints().forEach(point => point.select(false));
   let lastInputHash: string;
 
+  const updateGifLinks = (fen: Fen) => {
+    positionGifLink.href = xhrUrl(`/export/gif/${fen.replace(/ /g, '_')}`, {
+      color: ctrl.bottomColor(),
+      lastMove: ctrl.node.uci,
+      variant: ctrl.data.game.variant.key,
+      theme: document.body.dataset.boardTheme,
+      piece: document.body.dataset.pieceSet,
+    });
+    gameGifLink.href = xhrUrl(`/game/export/gif/${ctrl.bottomColor()}/${data.game.id}.gif`, {
+      theme: document.body.dataset.boardTheme,
+      piece: document.body.dataset.pieceSet,
+    });
+  };
+
   if (!window.LichessAnalyseNvui) {
+    lichess.pubsub.on('theme.change', () => updateGifLinks(inputFen.value));
+    lichess.pubsub.on('piece.change', () => updateGifLinks(inputFen.value));
```

Comment:
these pubsub listeners could also be avoided it seems

Reply:
If we did decide to go with this version of things, it did occur to me that we can at least combine these pubsubs into one.